metadata_backend/database/db_services.py

```diff
@@ -77,3 +74,58 @@ def read(db_service: DBService, collection: str,
             return db_service.database[collection].find(query)
         except errors.ConnectionFailure:
             raise
+
+    @staticmethod
+    def update(db_service: DBService, collection: str,
+               accession_id: str, data_to_be_updated: Dict) -> None:
+        """Update some elements of object by its accessionId.
+
+        :param db_service: Service that connects to MongoDB database
+        :param collection: Collection where document should be searched from
+        :param accession_id: Accession id for object to be updated
+        :param data_to_be_updated: JSON representing the data that should be
+        updated to object, can replace previous fields and add new ones.
+        :raises: Error when read fails for any Mongodb related reason
+        """
+        try:
+            find_by_id_query = {"accessionId": accession_id}
+            update_operation = {"$set": data_to_be_updated}
+            db_service.database[collection].update_one(find_by_id_query,
+                                                       update_operation)
+        except errors.ConnectionFailure:
+            raise
+
+    @staticmethod
+    def replace(db_service: DBService, collection: str,
+                accession_id: str, data_to_be_updated: Dict) -> None:
+        """Replace whole object by its accessionId.
+
+        :param db_service: Service that connects to MongoDB database
+        :param collection: Collection where document should be searched from
+        :param accession_id: Accession id for object to be updated
+        :param data_to_be_updated: JSON representing the data that replaces
+        old data
+        :raises: Error when read fails for any Mongodb related reason
+        """
+        try:
+            find_by_id_query = {"accessionId": accession_id}
+            db_service.database[collection].replace_one(find_by_id_query,
+                                                        data_to_be_updated)
+        except errors.ConnectionFailure:
```

Comment:
I think we should catch more that `ConnectionFailure`, maybe catch the specific ones first and then catch the general exception.

should be the same in `delete`.

also on `ConnectionFailure`, one should do a retry mechanism to retry the previous operation e.g. 10 times in 2-5 min after which raise an error (probably could be a later task -- add an issue if so).

Reply:
yeah, maybe we could catch ConnectionFailure, OperationFailure and then other pymongo errors.

Retry mechanism sounds like a task for later, I'll add issue for that.